cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/sidebar_filtering.spec.js

```diff
@@ -0,0 +1,303 @@
+    cy.getElementByTestId('sidebarPanel').should('not.be.visible');
+    dataExplorer.clickSidebarCollapseBtn(false);
+    cy.getElementByTestId('sidebarPanel').should('be.visible');
+  };
+
+  if (indexPattern) {
+    dataExplorer.selectIndexPatternDataset(INDEX_PATTERN_NAME, 'DQL');
+    dataExplorer.setQueryEditorLanguage('DQL');
+    collapseAndExpand();
+    dataExplorer.setQueryEditorLanguage('Lucene');
+    collapseAndExpand();
+  } else {
+    dataExplorer.selectIndexDataset(DATASOURCE_NAME, INDEX_NAME, 'PPL', 'timestamp');
+  }
+  dataExplorer.setQueryEditorLanguage('PPL');
+  collapseAndExpand();
+  dataExplorer.setQueryEditorLanguage('OpenSearch SQL');
+  collapseAndExpand(true);
+  if (indexPattern) {
+    dataExplorer.setQueryEditorLanguage('DQL');
+    cy.getElementByTestId('sidebarPanel').should('be.visible');
+  }
+};
+
+const checkCollapse = (indexPattern = true) => {
+  if (indexPattern) {
+    dataExplorer.selectIndexPatternDataset(INDEX_PATTERN_NAME, 'DQL');
+    dataExplorer.setQueryEditorLanguage('DQL');
+    dataExplorer.clickSidebarCollapseBtn();
+    cy.getElementByTestId('sidebarPanel').should('not.be.visible');
+    dataExplorer.setQueryEditorLanguage('Lucene');
+    cy.getElementByTestId('sidebarPanel').should('not.be.visible');
+  } else {
+    dataExplorer.selectIndexDataset(DATASOURCE_NAME, INDEX_NAME, 'PPL', 'timestamp');
+  }
+  dataExplorer.setQueryEditorLanguage('PPL');
+  if (!indexPattern) {
+    dataExplorer.clickSidebarCollapseBtn();
+  }
+  cy.getElementByTestId('sidebarPanel').should('not.be.visible');
+  dataExplorer.setQueryEditorLanguage('OpenSearch SQL');
+  cy.getElementByTestId('sidebarPanel').should('not.be.visible');
+  if (indexPattern) {
+    dataExplorer.setQueryEditorLanguage('DQL');
+    cy.getElementByTestId('sidebarPanel').should('not.be.visible');
+  }
+  dataExplorer.clickSidebarCollapseBtn(false);
+};
+
+describe('sidebar spec', () => {
+  before(() => {
+    // Load test data
+    cy.setupTestData(
+      SECONDARY_ENGINE.url,
+      ['cypress/fixtures/query_enhancements/data-logs-1/data_logs_small_time_1.mapping.json'],
+      ['cypress/fixtures/query_enhancements/data-logs-1/data_logs_small_time_1.data.ndjson']
+    );
+    // Add data source
+    cy.addDataSource({
+      name: `${DATASOURCE_NAME}`,
+      url: `${SECONDARY_ENGINE.url}`,
+      authType: 'no_auth',
+    });
+
+    // Create workspace
+    cy.deleteWorkspaceByName(`${WORKSPACE_NAME}`);
+    miscUtils.visitPage('/app/home');
+    cy.createInitialWorkspaceWithDataSource(`${DATASOURCE_NAME}`, `${WORKSPACE_NAME}`);
+    cy.wait(2000);
+    cy.createWorkspaceIndexPatterns({
+      url: `${BASE_PATH}`,
+      workspaceName: `${WORKSPACE_NAME}`,
+      indexPattern: 'data_logs_small_time_1',
+      timefieldName: 'timestamp',
+      indexPatternHasTimefield: true,
+      dataSource: DATASOURCE_NAME,
+      isEnhancement: true,
+    });
+    cy.navigateToWorkSpaceHomePage(`${BASE_PATH}`, `${WORKSPACE_NAME}`);
```

Comment:
```suggestion
    cy.navigateToWorkspaceSpecificPage({
        url: BASE_PATH, 
        workspaceName: WORKSPACE_NAME,
        page: 'discover',
        isEnhancement: true,
    });
```

Reasoning is that the `homepage` in neo is not the discover page

Reply:
Doesn't work for me. It gets stuck on this screen:
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/e0cfe5af-da67-4485-ab0e-cf43887aad4c" />

Reply:
Fixed